packages/aws-cdk/lib/context-providers/index.ts

```diff
@@ -64,7 +66,72 @@ export async function provideContextValues(
         ? await sdk.resolveEnvironment(environment)
         : { account: '?', region: '?', name: '?' };
 
-      const arns = await replaceEnvPlaceholders({
+      const arns = isSdkProviderV3(sdk)
+        ? await replaceEnvPlaceholdersv3({
+          lookupRoleArn: missingContext.props.lookupRoleArn,
+        }, resolvedEnvironment, sdk)
+        : await replaceEnvPlaceholders({
+          lookupRoleArn: missingContext.props.lookupRoleArn,
+        }, resolvedEnvironment, sdk);
+
+      value = await provider.getValue({ ...missingContext.props, lookupRoleArn: arns.lookupRoleArn });
+    } catch (e: any) {
+      value = { [cxapi.PROVIDER_ERROR_KEY]: e.message, [TRANSIENT_CONTEXT_KEY]: true };
+    }
+    context.set(key, value);
+    debug(`Setting "${key}" context to ${JSON.stringify(value)}`);
+  }
+}
+
+// Type guard to check if the sdk is SdkProviderv3
+function isSdkProviderV3(sdk: SdkProvider | SdkProviderv3): sdk is SdkProviderv3 {
+  return 'pluginsv3' in sdk;
+}
+
+/**
+ * Iterate over the list of missing context values and invoke the appropriate providers from the map to retrieve them
+ */
+export async function provideContextValuesv3(
```

Comment:
Where is this used?

Reply:
it was used previously, I don't know if I still need it yet lol. I'll be sure to get rid of it before I pull it out of draft. I'm just keeping it around for safe keeping for now